src/main/java/core/basesyntax/strategy/ReturnHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class ReturnHandler implements OperationHandler {
+    private static final String RETURN = "r";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(RETURN)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
+        return listOfFruits;
```

Comment:
The method `getFruitFinalQuantity` in `ReturnHandler` does not seem to follow the Strategy pattern correctly. Each handler should be responsible for a single type of operation. This method also appears to be mutating the input list `listOfFruits`, which could lead to unexpected behavior. Instead, consider creating a new list with the updated quantities or refactor the method to avoid side effects.